src/pin.py

```diff
@@ -0,0 +1,56 @@
+from hashlib import hmac_sha512, sha256
+from struct import unpack
+from bitcoin import bip39
+from platform import storage_root
+from rng import get_random_bytes
+from ubinascii import hexlify, unhexlify
+import ujson as json
+
+secret_fname = "%s/%s" % (storage_root, "secret.json")
+
+def antiphishing_word(pin_digit):
+    _hmac = hmac_sha512(Secret.encryption_key, pin_digit)
+    hmac_num = unpack('<H', bytearray(_hmac[0:8]))[0]
+    idx = hmac_num % len(bip39.WORDLIST)
+    word = (bip39.WORDLIST[idx])
+    return word
+
+class Secret:
+    # FIXME key should be read from a location which was written to by the bootloader
+    encryption_key = None
+    hmac_key       = None
+    hmac           = None
+
+    @staticmethod
+    def load_secret():
```

Comment:
Here it would be better if we derive the decryption secret from a secret stored in `secret.json` file AND the pin. 
Then if the attacker tries to glitch the microcontroller to pass the pin check he would end up with a wrong decryption secret.